libraries/serialization-jackson/base/src/test/java/io/kiota/serialization/json/CustomJsonParseNodeTests.java

```diff
@@ -0,0 +1,160 @@
+        }
+
+        public void setInteger(Integer integer) {
+            this.integer = integer;
+        }
+
+        public Integer getInteger2() {
+            return integer2;
+        }
+
+        public void setInteger2(Integer integer2) {
+            this.integer2 = integer2;
+        }
+
+        public OffsetDateTime getDateTime() {
+            return dateTime;
+        }
+
+        public void setDateTime(OffsetDateTime dateTime) {
+            this.dateTime = dateTime;
+        }
+
+        @jakarta.annotation.Nonnull
+        public static TestObject createFromDiscriminatorValue(
+                @jakarta.annotation.Nonnull final ParseNode parseNode) {
+            Objects.requireNonNull(parseNode);
+            return new TestObject();
+        }
+
+        @Override
+        public Map<String, Consumer<ParseNode>> getFieldDeserializers() {
+            final HashMap<String, java.util.function.Consumer<ParseNode>> deserializerMap =
+                    new HashMap<String, java.util.function.Consumer<ParseNode>>(9);
+            deserializerMap.put(
+                    "integer",
+                    n -> {
+                        this.setInteger(n.getIntegerValue());
+                    });
+            deserializerMap.put(
+                    "integer2",
+                    n -> {
+                        this.setInteger2(n.getIntegerValue());
+                    });
+            deserializerMap.put(
+                    "dateTime",
+                    n -> {
+                        this.setDateTime(n.getOffsetDateTimeValue());
+                    });
+            return deserializerMap;
+        }
+
+        @Override
+        public void serialize(SerializationWriter writer) {
+            throw new UnsupportedOperationException();
+        }
+    }
+
+    @Test
+    void testIntegerConversion() throws IOException {
+        class DivideEvenIntegersJsonParseNode extends JsonParseNode {
+            DivideEvenIntegersJsonParseNode(JsonParseNodeFactory factory, JsonNode currentNode) {
+                super(factory, currentNode);
+            }
+
+            @Override
+            public Integer getIntegerValue() {
+                Integer value = super.getIntegerValue();
+
+                if (value != null && value % 2 == 0) {
+                    value = value / 2;
+                }
+
+                return value;
+            }
+        }
+
+        JsonParseNodeFactory factory =
+                new JsonParseNodeFactory() {
+                    @Override
+                    protected JsonParseNode createJsonParseNode(JsonNode currentNode) {
+                        return new DivideEvenIntegersJsonParseNode(this, currentNode);
+                    }
+                };
+
+        JsonParseNode node =
+                factory.createJsonParseNode(mapper.readTree("{\"integer\":7,\"integer2\":42}"));
+
+        var objValue = node.getObjectValue(TestObject::createFromDiscriminatorValue);
```

Comment:
uhm, this is interesting, @MikeEdgar have you attempted to use this mechanism in a full fledged Kiota client?

I'm interested to see an E2E example on how to use the functionality to patch a server response, I'm not 100% sure how you envision a full, working, integration.

Reply:
Thanks a lot @MikeEdgar !

Uh, I see, so the mechanism works by overriding the behavior of a ser/deser that is already being dispatched.